### ISSUE TITLE
Raise an error for non int16 encoded .wav files with kaldiio feature extraction

### ISSUE THREAD

**Issue by popcornell** [espnet/espnet#4214]
Currently there is no check for making sure that .wav audio files are encoded to int16 PCM. 
This can cause a lot of headaches when Kaldi-based feature extraction (through `kaldiio`) is used as it assumes that the file is encoded in int16 PCM.
No error is raised while it should ve been since the waveform data will be a total mess (wrongly decoded) and the model will fail if it has been trained on int16 PCM .wav files. 

***
Why does this matter ? 
**Torchaudio does not encode to the standard int16 PCM by default if the input is a float32 tensor (almost always) !** see https://pytorch.org/audio/stable/backend.html#backend
***

I'll try to reproduce this tomorrow. Raising for now. 

@simpleoier can confirm this.

**Comment by simpleoier** [comment]
Right. [Here](https://github.com/espnet/espnet/blob/16f2cd333a994e49fc8fb69b32a1894f87e51a13/egs2/TEMPLATE/asr1/pyscripts/audio/format_wav_scp.py#L89) is a note, but not easy to find. 
`# Note: kaldiio supports only wav-pcm-int16le file.`

**Comment by kamo-naoyuki** [comment]
I see, thanks, it is a good suggestion. I'll consider it.

**Comment by popcornell** [comment]
this correctly throws an error however. 
@simpleoier can you point to the exact recipe you were using ?

kaldiio version: '2.17.2'

```python
from kaldiio import ReadHelper
import torchaudio
import soundfile as sf
import torch
import numpy as np

filepath = "/tmp/dummy.wav"
wavscp_path = "/tmp/wav.scp"
fs = 16000
dummydata = torch.randint(0, 1, (1, fs)).float()
#sf.write(filepath, dummydata[0].numpy(), fs)
torchaudio.save(filepath, dummydata, fs)

# we write wav.scp
with open(wavscp_path, "w") as f:
    f.writelines("dummy {}".format(filepath))


from_kaldiio = []

with ReadHelper('scp:{}'.format(wavscp_path)) as reader:
    for key, (rate, numpy_array) in reader:
        from_kaldiio.append(numpy_array)

np.testing.assert_array_almost_equal(from_kaldiio[0], dummydata[0].numpy())
```
---
Error is `ValueError: bytes_per_sample must be 1, 2, 4 or 8`
